### PR TITLE
Add personal workspace support

### LINKED ISSUE #4768: Add support for personal workspaces
A personal workspace will be one whose name matches a defined pattern (such as "Drive"). These workspaces should be mounted in a specific directory in order to visually distinguish them from the standard workspaces.

The following options could be added to Parsec:
- `personal_workspace_base_dir`: base dir to mount personal workspaces
- `personal_workspace_name_pattern`: naming pattern to define personal workspaces

This was initially proposed for the Resana-Secure project: https://github.com/Scille/resana-secure/issues/183

### PR DESCRIPTION
**Pull request by mmmarcos**
## Describe your changes

This PR adds support for personal workspaces (see #4768). 

Closes #4768

## Checklist

Before you submit this pull request, please make sure to:

- [x] Keep changes in the pull request as small as possible
- [x] Ensure the commit history is sanitized
- [x] Give a meaningful title to your PR
- [x] Describe your changes
- [x] Link any related issue in the description

If this PR adds a new feature or fixes a bug:

- [x] I have added or updated relevant tests

### CODE REVIEW

File: tests/core/mountpoint/test_base.py [hunk 1]
```diff
@@ -674,3 +674,100 @@ async def _in_trio_land(user_fs, mountpoint_manager):
...
+
+    # Create standard and personal workspaces
+    workspaces = {}
+    for name in standard_workspaces + personal_workspaces:
+        workspaces[name] = await alice_user_fs.workspace_create(EntryName(name))
+
+    await alice_user_fs.sync()
+
+    # Add some files
+    for w in workspaces:
```
Comment (touilleMan) [inline tests/core/mountpoint/test_base.py:694]:
```suggestion
    for workspace_name, workspace_fs in workspaces.items():
```

File: tests/core/mountpoint/test_base.py [hunk 2]
```diff
@@ -674,3 +674,100 @@ async def _in_trio_land(user_fs, mountpoint_manager):
...
+    # In the following scenarios, workspaces should be mounted on
+    # the standard base mountpoint
+
+    async with mountpoint_manager_factory(
+        alice_user_fs,
+        alice_user_fs.event_bus,
+        standard_base_mountpoint,
+        mountpoint_in_directory=True,
+        # Mount without specifying a path for personal workspaces
+        # personal_workspace_base_path=personal_base_mountpoint,
```
Comment (touilleMan) [inline tests/core/mountpoint/test_base.py:753]: dead code ?

**Review by touilleMan — CHANGES_REQUESTED** [review]

File: parsec/core/config.py [hunk 3]
```diff
@@ -61,6 +61,8 @@ class CoreConfig:
 
     mountpoint_enabled: bool = False
     mountpoint_in_directory: bool = False
+    personal_workspace_base_dir: Path | None = None
```
[the remark this answers is not in the document]
Comment (mmmarcos) [reply parsec/core/config.py:67]: done

File: parsec/core/mountpoint/manager.py [hunk 4]
```diff
@@ -97,6 +100,14 @@ def __init__(
         self._nursery = nursery
         self._mountpoint_tasks: dict[tuple[EntryID, DateTime | None], TaskStatus[Any]] = {}
         self._timestamped_workspacefs: dict[EntryID, dict[DateTime, WorkspaceFSTimestamped]] = {}
+        self.personal_workspace_base_path = (
+            personal_workspace_base_path
```
[the remark this answers is not in the document]
Comment (mmmarcos) [reply parsec/core/mountpoint/manager.py]: done

File: tests/core/mountpoint/test_base.py (hunk 1 above)
Comment (mmmarcos) [reply tests/core/mountpoint/test_base.py]: done

File: tests/core/mountpoint/test_base.py (hunk 2 above)
Comment (mmmarcos) [reply tests/core/mountpoint/test_base.py]: It was just to emphasize that this parameter is not being used. But I agree that it is not very clear. I've removed the dead code.

File: parsec/core/mountpoint/manager.py [hunk 5]
```diff
@@ -383,11 +405,19 @@ async def mountpoint_manager_factory(
...
+    personal_workspace_base_path: Path | None = None,
+    personal_workspace_name_pattern: str | None = None,
 ) -> AsyncGenerator[MountpointManager, Any]:
     config = {"debug": debug}
 
     runner = get_mountpoint_runner()
 
+    # In order to mount personal workspaces into a directory, the
+    # mountpoint_in_directory option should be enabled, otherwise the default
+    # base mountpoint path is used
```
[the remark this answers is not in the document]
Comment (mmmarcos) [reply parsec/core/mountpoint/manager.py]: Agree with the first part (I added asserts with comments/messages). For the second part, the `base_mountpoint_path` parameter is already declared as `base_mountpoint_path: Path`, I would prefer not to change that, to minimize the risk of introducing a regression (although it should not be high...)

**Review by touilleMan — APPROVED** [review]

**Pull request merged**